src/TaskList.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+import firebase from './firebase';
+
+
+const TaskList = () => {
+    const [taskList, setTaskList ] = useState();
+    useEffect(() => {
+        const tlRef = firebase.database().ref('Task');
+        tlRef.on('value', (upcoming) => {
+            const task = upcoming.val();
+            const taskList = []
+            for (let id in task) {
+                taskList.push({ task });
```

Comment:
I'm not familiar enough with React to be sure, but I think you might be redefining taskList here, or trying to change the state value manually?